com.ibm.streamsx.topology/com.ibm.streamsx.topology.functional.python/PyFunctionMultiTransform/PyFunctionMultiTransform_cpp.cgt

```diff
@@ -0,0 +1,187 @@
+%>
+
+    PyObject * appFunction = 
+      streamsx::topology::Splpy::loadFunction(<%=$module%>, <%=$functionName%>);
+
+<% if ($pystyle eq 'python') { %>
+
+    PyObject * depickleInput = 
+      streamsx::topology::Splpy::loadFunction("streamsx.topology.runtime", "depickleInput");
+
+    PyObject * funcArg = PyTuple_New(1);
+    PyTuple_SetItem(funcArg, 0, appFunction);
+
+    function_ = PyObject_CallObject(depickleInput, funcArg);
+    // TODO error check
+    Py_DECREF(depickleInput);
+    Py_DECREF(funcArg);
+    
+    iterableObjectFunction_ = 
+      streamsx::topology::Splpy::loadFunction("streamsx.topology.runtime", "iterableObject");
+
+<%}%>
+<% if ($pystyle eq 'string') { %>
+    function_ = appFunction;
+<%}%>
+
+    PyGILState_Release(gstate);
+}
+
+// Destructor
+MY_OPERATOR::~MY_OPERATOR() 
+{
+    // Finalization code goes here
+    if (function_) {
+      Py_DECREF(function_);
+    }
+    if (iterableObjectFunction_) {
+      Py_DECREF(iterableObjectFunction_);
+    }
+}
+
+// Notify port readiness
+void MY_OPERATOR::allPortsReady() 
+{
+}
+ 
+// Notify pending shutdown
+void MY_OPERATOR::prepareToShutdown() 
+{
+    // This is an asynchronous call
+}
+
+// Processing for source and threaded operators   
+void MY_OPERATOR::process(uint32_t idx)
+{
+}
+
+// Tuple processing for mutating ports 
+void MY_OPERATOR::process(Tuple & tuple, uint32_t port)
+{
+}
+
+// Tuple processing for non-mutating ports
+void MY_OPERATOR::process(Tuple const & tuple, uint32_t port)
+{
+  IPort0Type const &ip = static_cast<IPort0Type const &>(tuple);
+
+<% if ($pystyle eq 'python') { %>
+  SPL::blob value = ip.get___spl_po();
+<%}%>
+
+<% if ($pystyle eq 'string') { %>
+  SPL::rstring value = ip.get_string();
+<%}%>
+
+  std::auto_ptr<SPL::blob> pBlob;
+  PyGILState_STATE gstate;
+  gstate = PyGILState_Ensure();
+
+  // invoke python application function and get return value
+  PyObject * pyBytes  = streamsx::topology::Splpy::pyBlobToBytes(value);
+  PyObject * pyReturnVar = streamsx::topology::Splpy::pyTupleFunc(function_, pyBytes);
+
+  if (pyReturnVar == Py_None){
+    Py_DECREF(pyReturnVar);
+    PyGILState_Release(gstate);
+    return;
+  } else if(pyReturnVar == 0){
+    PyErr_Print();
+    PyGILState_Release(gstate);
+    throw;
+  } 
+
+  // change python return value into an iterator function
+  PyObject * pyIterator = streamsx::topology::Splpy::pyTupleFunc(iterableObjectFunction_, pyReturnVar);
+  
+  while (!getPE().getShutdownRequested()) {
+    // call next() on the iterator, and get the pickled return value
+    PyObject * pyPickledReturnVar = PyObject_CallObject(pyIterator, NULL);
+    if (pyPickledReturnVar == Py_None){
+      Py_DECREF(pyPickledReturnVar);
+      Py_DECREF(pyIterator);
+      break;
+    } else if(pyPickledReturnVar == 0){
+      PyErr_Print();
+      Py_DECREF(pyIterator);
+      PyGILState_Release(gstate);
+      throw;
+    }
+
+    // construct spl blob

```

Comment:
Pickling of Python of objects results in blobs, so I am using PyBytes_Size() / PyBytes_AsString() to construct tuples.  This is consistent with how the source operator constructs SPL tuples from pickled values.
Other than the source operator, the only other place I see SPL tuples being created is py_functionReturnToTyples.cgt.  It is used in specific cases - the return type must be tuple or list, and it works with primitive times.  
The other functional operators like source, sink, filter work with tuples as blobs, so having multiTransform use blobs seemed appropriate.